crates/zkvm/adapters/native/src/host.rs

```diff
@@ -0,0 +1,60 @@
+use std::{fmt, sync::Arc};
+
+use strata_zkvm::{Proof, ProofType, VerificationKey, ZkVmHost, ZkVmInputBuilder, ZkVmResult};
+
+use crate::{input::NativeMachineInputBuilder, zkvm::NativeMachine};
+
+type ProcessProofFn = dyn Fn(&NativeMachine) -> ZkVmResult<()> + Send + Sync;
+
+#[derive(Clone)]
+pub struct NativeHost {
+    pub process_proof: Arc<Box<ProcessProofFn>>,
+}
+
+impl ZkVmHost for NativeHost {
+    type Input<'a> = NativeMachineInputBuilder;
+
+    fn prove<'a>(
+        &self,
+        prover_input: <Self::Input<'a> as ZkVmInputBuilder<'a>>::Input,
+        _proof_type: ProofType,
+    ) -> ZkVmResult<(Proof, VerificationKey)> {
+        (self.process_proof)(&prover_input)?;
+        let output_ref = prover_input.output.borrow();
+
+        let output = if output_ref.is_empty() {
+            vec![]
+        } else {
+            output_ref[0].clone()
```

Comment:
Why are we indexing the first element only `output_ref[0]` ?
Will it fail if there were multiple outputs ?

Reply:
Updated this so that it only has a single output. This is similar to [Risc0 journal](https://github.com/risc0/risc0/blob/main/risc0/zkvm/src/receipt.rs#L274) and [SP1 public values buffer](https://github.com/succinctlabs/sp1/blob/dev/crates/primitives/src/io.rs#L9). 

Since all our guest programs uses a single output, this should be fine for now. In case we need multiple outputs, we can revisit this.